### PR TITLE
GN-3998: Instead of joining into 'block_rdfa' node - select it when deleting at the end of it

### PR DESCRIPTION
**Pull request by dkozickis**
https://user-images.githubusercontent.com/769698/229424296-8a94810e-f42f-476e-b62f-21c11594f4e5.mp4

### CODE REVIEW

**Review by lagartoverde — APPROVED** [review]
Looks good :)

**Review by abeforgit — CHANGES_REQUESTED** [review]
For some reason this makes every block behave like the block_rdfa
I don't think there's anything wrong with your code but there's some interaction going on with the command chain I think.

fyi non-text nodes are selectable by default

**Comment by lagartoverde** [comment]
> For some reason this makes every block behave like the block_rdfa I don't think there's anything wrong with your code but there's some interaction going on with the command chain I think.
> 
> fyi non-text nodes are selectable by default

Could you elaborate on this? Because I didn't see anything wrong and want to know what's wrong for the next one :)

**Comment by abeforgit** [comment]
additionally this pr doesn't deal with delete. I'm fine with splitting that into another pr though, lets get this one consistent first

**Comment by abeforgit** [comment]
the point is that the two examples: the list item and the first paragraph, are not block_rdfa nodes but they are still nodeselected

**Comment by abeforgit** [comment]
unfortunately....
minimal repro:
clear the doc, use customhtml to insert:
```html
<div typeof="Sometype"><p>this is inside the rdfa node</p></div><p><br>text</p>
```
backspace away the "text" and keep backspacing, now it doesn't trigger the nodeselection
but if you click on the newline right before the rdfa node, backspacing _does_ trigger correctly

**Comment by abeforgit** [comment]
interestingly, you get p much the same behavior if you disable your selectBlockRdfaNode command altogether
I suspect you're approaching the same implementation as the selectNodeBackward command which is already a part of the backspace chain

**Comment by abeforgit** [comment]
Im actually quite surprised at that, it breaks my assumptions about "isolating"

**Comment by abeforgit** [comment]
aah further clues: the culprit actually seems to be this ad-hoc construction in the backspace chain:
```ts
  (state, dispatch, view) => {
    const isInTable = hasParentNodeOfType(state.schema.nodes.table)(
      state.selection
    );
    if (joinBackward(state, dispatch) && dispatch && view) {
      const { state } = view;
      if (!isInTable) {
        selectParentNodeOfType(state.schema.nodes.table)(state, dispatch, view);
      }
      return true;
    }
    return false;
  },
```

If I move the selectNodeBackward command to BEFORE this construction, it works as I had imagined

so I think it's the "joinBackward" mentioned above seems to have more power than we thought
the assumption was that it would follow the same rules as the "default" backspace behavior when it came to isolating/selectable nodes. It seems it maybe doesn't? Not sure, but I'd investigate there for sure

**Review by abeforgit — CHANGES_REQUESTED** [review]
I couldn't break it anymore, good job :) 
The logic is ready, but it needs to be opt-in. So rather simply export the command and add an explanation in the docs how to enable it

File: addon/core/keymap.ts [hunk 1]
```diff
@@ -79,7 +94,7 @@ const del = chainCommands(
 /// * **Delete** and **Mod-Delete** to `deleteSelection`, `joinForward`, `selectNodeForward`
 /// * **Mod-Delete** to `deleteSelection`, `joinForward`, `selectNodeForward`
 /// * **Mod-a** to `selectAll`
-export const pcBaseKeymap: Keymap = (schema: Schema) => ({
+export const pcBaseKeymap: Keymap = (schema, options) => ({
```
Comment (abeforgit) [inline addon/core/keymap.ts:97]: its ok for now, I suspect we'll revisit the outside-facing interface when we actually turn this on

File: addon/nodes/block-rdfa.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ export const block_rdfa: NodeSpec = {
     __tag: { default: 'div' },
   },
   defining: true,
+  isolating: true,
```
Comment (abeforgit) [inline addon/nodes/block-rdfa.ts:12]: unfortunately this also breaks current behavior
I suggest simply exporting a separate variation of this node, or just documenting how to extend nodes the host might want to treat like this.
we're in the early phases of the configuration interface where we simply expose all wires and ask the consumer to plug them in, so I'm ok with just documenting

**Review by abeforgit — CHANGES_REQUESTED** [review]
one more thing to do to avoid breaking the public api
also, I would like a page in the dummy app that has this active so we can start playing around with it (so a new page, so we can easily compare)

**Comment by abeforgit** [comment]
asking for the extra page is kind of out of scope, depending on how you look at it, so feel free to make a ticket for it and not do it in this pr
but it does have to be made non-breaking for current consuming apps

**Pull request merged**